js/private/image/index.ts

```diff
@@ -0,0 +1,258 @@
+import { createReadStream, createWriteStream } from 'node:fs'
+import { readdir, readFile, realpath, stat } from 'node:fs/promises'
+import * as path from 'node:path'
+import { Readable, Stream } from 'node:stream'
+import { pathToFileURL } from 'node:url'
+import { createGzip } from 'node:zlib'
+import { pack, Pack } from 'tar-stream'
+
+const MTIME = new Date(0)
+
+type HermeticStat = {
+    mtime: Date
+    mode: number
+    size?: number
+}
+
+type Entry = {
+    is_source: boolean
+    is_directory: boolean
+    dest: string
+    root?: string
+    remove_non_hermetic_lines?: boolean
+}
+type Entries = { [path: string]: Entry }
+
+type Compression = 'gzip' | 'none'
+
+function findKeyByValue(entries: Entries, value: string): string {
+    for (const [key, { dest: val }] of Object.entries(entries)) {
+        if (val == value) {
+            return key
+        }
+    }
+    throw new Error(
+        `couldn't map ${value} to a path. please file a bug at https://github.com/aspect-build/rules_js/issues/new/choose`
```

Comment:
nit: try to print at least the beginning of `Entries` here as well, or else such a filed bug will be hard for us to do anything with?